nntrainer/models/neuralnet.cpp

```diff
@@ -774,6 +782,11 @@ int NeuralNetwork::train_run(std::function<bool(void *userdata)> stop_cb) {
                 << " - Training Loss: " << stat.loss;
       ml_logi("# %d / %d - Training Loss: %f", epoch_idx, getEpochs(),
               stat.loss);
+      ml_logd("[NNTrainer] Training epoch %d / %d finished successfully.",
+              epoch_idx, getEpochs());
+    } else {
+      ml_logd("[NNTrainer] Training stopped unexpectedly during epoch %d.",
+              epoch_idx);
```

Comment:
I wonder why it is "unexpectedly"

Reply:
Changed to "stopped by stop callback function"

Reply:
Great.